src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -1006,9 +1006,16 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         // The `visit_expr` ensures it resolves as an expression
                         // which must be a `Var_t` pointing to a `Variable_t`,
                         // so no checks are needed:
+                        ImplicitCastRules::set_converted_value(al, x.base.base.loc, &value,
+                                                ASRUtils::expr_type(value), ASRUtils::expr_type(object));
+                        ASR::expr_t* value_value = ASRUtils::expr_value(value);
```

Comment:
```suggestion
                        ASR::expr_t* expression_value = ASRUtils::expr_value(value);
```
etc.
Just to have it a little more readable.